lab_1_classify_by_unigrams/main.py

```diff
@@ -3,31 +3,34 @@
 Language detection
 """
 
-
 def tokenize(text: str) -> list[str] | None:
-    """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation, digits and other symbols
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
+    if isinstance(text, str):
+        tokens = []
+        text = text.lower()
+        for i in text:
+            if i.isalpha():
+                tokens.append(i)
+        return tokens
+    else:
+        return None
 
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens
-    :return: a dictionary with frequencies
-    """
+    if isinstance(tokens, list) and all(isinstance(el, str) for el in tokens):
+        frequency_dict = {el: (tokens.count(el) / len(tokens)) for el in tokens}
+        return frequency_dict
+    else:
```

Comment:
If you, however, decide not to do an early return, can you explain why you need `else` here?